src/components/Quiz.jsx

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect } from "react";
+import { QUESTIONS } from "../quizQuestions";
+import SummaryScreen from "./SummaryScreen";
+import WelcomeScreen from "./WelcomeScreen";
+import QuestionCard from "./QuestionCard";
+
+const Quiz = () => {
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [score, setScore] = useState(0);
+  const [isQuizFinished, setIsQuizFinished] = useState(false);
+  const [answerListArr, setAnswerListArr] = useState([]);
+  const [showsWelcomeScreen, setShowWelcomeScreen] = useState(true);
+
+  const handleAnswerClick = (selectedAnswer) => {
+    setAnswerListArr((prevAnswerListArr) => [
+      ...prevAnswerListArr,
+      {
+        question: QUESTIONS[currentQuestionIndex].text,
```

Comment:
Knowing that we pass "QUESTIONS" to "SummaryScreen," the only value that should be stored is the index of the selected answer for a given task.